src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Resources/Bundle/BundleHandlerTests.cs

```diff
@@ -192,6 +207,72 @@ public async Task GivenABundle_WhenMultipleRequests_ReturnsABundleResponseWithCo
             Assert.Equal("200", bundleResource.Entry[2].Response.Status);
         }
 
+        [Fact]
+        public async Task GivenATransactionBundleWithConditionalReferences_WhenResolved_ThenReferencesValuesAreUpdatedCorrectly()
```

Comment:
Are we testing the `BundleHandler` or the `TransactionBundleValidator` here? Would it make sense to make the `TransactionBundleValidator` mockable instead? Thoughts?

Reply:
As per our discussion l am leaving it as is.